### PR TITLE
Modify Project Structure

### PR DESCRIPTION
**Pull request by joel-putnam**
# Restructure Project Files

This modifies the project structure to allow for easily adding functions to convert Rhino objects into Blender equivalents as well as facilitates the creation of a typical blender add on.

## Detailed Explanation

- Convert Project to a traditional Blender add on allowing for installation as a zip file
- Add in the concept of a "Converter" to convert Rhino objects into Blender equivalents such as mesh/curve/camera etc. 
- Converters can be added by creating the appropriate function in a new file and adding the converter to the RHINO_TYPE_TO_IMPORT dictionary  found in the converters sub-module 
- Allows for this add on to be written and debugged utilizing the Visual Studio Code Blender Extension found here: https://github.com/JacquesLucke/blender_vscode

### CODE REVIEW

File: import_3dm/read3dm.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. """
+
+import os.path
+import bpy
+import rhino3dm as r3d
+from .converters import *
```
Comment (jesterKing) [inline import_3dm/read3dm.py]: No blanket * import

File: import_3dm/read3dm.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    materials = {}
+
+    handle_materials(context, model, materials)
+
+    handle_layers(context, model, toplayer, layerids, materials )
+        
+    for ob in model.Objects:
+        og=ob.Geometry
+        if og.ObjectType not in RHINO_TYPE_TO_IMPORT: continue
+        exporter = RHINO_TYPE_TO_IMPORT[og.ObjectType]
```
Comment (jesterKing) [inline import_3dm/read3dm.py]: I suggest the name  "convert_rhino_object". We aren't exporting, but importing. Conversion is a nice middle road, that you already nicely hinted at with the converters submodule

```suggestion
        convert_rhino_object = RHINO_TYPE_TO_IMPORT[og.ObjectType]
```

File: import_3dm/read3dm.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    
+        layeruuid = rhinolayer.Id
+        rhinomatname = rhinolayer.Name + "+" + str(layeruuid)
+        if matname:
+            rhinomat = materials[matname]
+        else:
+            rhinomat = materials[rhinomatname]
+        layer = layerids[str(layeruuid)][1]
+
+        exporter(og, context, n, attr.Name, attr.Id, layer, rhinomat)
```
Comment (jesterKing) [inline import_3dm/read3dm.py]:
```suggestion
        convert_rhino_object(og, context, n, attr.Name, attr.Id, layer, rhinomat)
```

**Comment by jesterKing** [comment]
Oh, and before I forget: do fix up the PR message - you left the template pretty much untouched (:

File: import_3dm/converters/layers.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. """
+
+from bpy_extras.node_shader_utils import PrincipledBSDFWrapper
+import rhino3dm as r3d
+from .utils import *
```
[the remark this answers is not in the document]
Comment (joel-putnam) [reply import_3dm/converters/layers.py]: I believe this is now addressed

File: import_3dm/__init__.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
+""" MIT License
```
[the remark this answers is not in the document]
Comment (jesterKing) [reply import_3dm/__init__.py]: Not quite, it is still a string. A comment block would be each line starting with a pound sign instead.

```
# MIT License
# text here
# and so on
```

**Comment by jesterKing** [comment]
Only the license text matter remains, the rest of the changes look good to me.

File: import_3dm/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (joel-putnam) [reply import_3dm/__init__.py]: Fixed now

**Pull request merged**